reports_edit.php

```diff
@@ -59,6 +59,17 @@
 
 
 
+// Clean up report data for deleted projects.
+$moduleName = preg_replace( '_v[0-9.]+$', '', $module->getModuleDirectoryName() );
+$module->query( 'DELETE FROM redcap_external_module_settings ' .
+                'WHERE external_module_id = (SELECT external_module_id FROM' .
+                ' redcap_external_modules WHERE directory_prefix = ?) AND project_id IS NULL ' .
+                "AND `key` REGEXP '^p[1-9][0-9]*-' AND `key` NOT REGEXP concat('^p(', " .
+                "(SELECT group_concat(project_id SEPARATOR '|') FROM redcap_projects), ')-') " .
```

Comment:
Does this delete the settings as soon as the project is deleted?  Just wondered as I have restored deleted projects before now, and it would be a shame to lose their settings at the 'initial delete' stage (where we can still get them back)

Reply:
This deletes the reports for projects which do not exist in the *redcap_projects* table. If a project is deleted but restorable, it will still exist in this table (flagged for deletion) until it is permanently deleted, so the reports are only deleted when the project is *permanently* deleted.